pkg/liveshare/rpc.go

```diff
@@ -40,7 +42,60 @@ func (r *rpcClient) do(ctx context.Context, method string, args, result interfac
 	return waiter.Wait(waitCtx, result)
 }
 
-type nullHandler struct{}
+type handlerFn func(conn *jsonrpc2.Conn, req *jsonrpc2.Request)
 
-func (nullHandler) Handle(ctx context.Context, conn *jsonrpc2.Conn, req *jsonrpc2.Request) {
+type requestHandler struct {
+	handlersMu sync.RWMutex
```

Comment:
RWMutex is overkill; Mutex is fine for a critical section this small.

Reply:
changed in latest commit